pub-matic-htb.js

```diff
@@ -462,21 +462,23 @@ function PubMaticHtb(configs) {
             }
 
             for (var i = 0; i < bids.length; i++) {
-                sizes = curReturnParcel.xSlotRef.sizes[0];
-
-                /**
-                 * This section maps internal returnParcels and demand returned from the bid request.
-                 * In order to match them correctly, they must be matched via some criteria. This
-                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
-                 * key to a key that represents the placement in the configuration and in the bid responses.
-                */
-                if (bids[i].impid === curReturnParcel.htSlot.getId()) {
-                    if (parseInt(bids[i].w, 10) === parseInt(sizes[0], 10)
-                        && parseInt(bids[i].h, 10) === parseInt(sizes[1], 10)) {
-                        curBid = bids[i];
-                        bids.splice(i, 1);
-
-                        break;
+                for (var index = 0; index < curReturnParcel.xSlotRef.sizes.length; index++) {
+                    sizes = curReturnParcel.xSlotRef.sizes[index];
+
+                    /**
+                     * This section maps internal returnParcels and demand returned from the bid request.
+                     * In order to match them correctly, they must be matched via some criteria. This
+                     * is usually some sort of placements or inventory codes. Please replace the someCriteria
+                     * key to a key that represents the placement in the configuration and in the bid responses.
+                    */
+                    if (bids[i].impid === curReturnParcel.htSlot.getId()) {
+                        if (parseInt(bids[i].w, 10) === parseInt(sizes[0], 10)
+                            && parseInt(bids[i].h, 10) === parseInt(sizes[1], 10)) {
+                            curBid = bids[i];
+                            bids.splice(i, 1);
+
+                            break;
```

Comment:
In the original code, we were running a single loop on bids, we were breaking the loop on finding the required impid.
Now we are running loop on bids and then internally on curReturnParcel.xSlotRef.sizes and breaking loop of sizes on finding impid, and loop on bids is continued. Is it the right behavior?

Reply:
i have updated the relevant piece of code.

Reply:
cool